src/docs/markdown/build.md

```diff
@@ -4,10 +4,24 @@ title: "Build from source"
 
 # Build from source
 
+There are 2 ways to build Caddy, as opposed to installation of pre-built executable:
+- Build from Git repo
+- Build using `xcaddy`
+
```

Comment:
I think we should probably add a section about building with Docker in here. It's a quite common need, no reason we don't just have this in our docs. We have it in https://hub.docker.com/_/caddy but that's not where people usually look I think. See the "Adding custom Caddy modules" section there.

Reply:
I copy-pasted that section from Docker Hub into a section on this page. Good?